Make XDMA driver compatible with linux kernel >= 5.16

leragequit opened the pull request:
I was having trouble getting the driver to compile on my raspberry pi

with this patch its working again. adjusted caio->iocb->ki_complete call to the new function signature if compiled on kernel >= 5.16

and switched the deprecated pci_unmap_sg to the new dma_unmap_sg as well as the set_dma_mask functions

Commit:
Adjusted driver to work with kernel 6.1

Commit:
make it compatible with kernel < 5.16

alonbl commented:
You should probably condition this with the kernel version to keep backward compatibility

Eaxamples available in code:

```
#if KERNEL_VERSION(4, 13, 0) <= LINUX_VERSION_CODE
```

leragequit commented:
Hey @alonbl ,

unfortunately I forgot all about this and don't have a 4.13 system to test right now. Is this something you can do trivially or should i try to look into this at some point in the future when more time is available?

Best regards,

Sascha

alonbl commented:
Fixed with commit 9cdc9e335d51dc4e0e9d9fd6b4592ae55e30de72

neutronstriker committed:
Update cdev_sgdma.c

missing ";" at line 128 after "caio->iocb->ki_complete(caio->iocb, numbytes, -EBUSY)"

leragequit committed:
Update readme.txt

added depmod step

leragequit committed:
Adjusted xmda driver to conform with new vm_flags visibily, class_create removal of module name and obsolete iov_iter->__iov

leragequit committed:
Use proper access method for iov

leragequit committed:
Merge pull request #1 from neutronstriker/master

Update cdev_sgdma.c

leragequit committed:
compatible with 6.1.x and 6.6.x now

leragequit committed:
Update README.md

leragequit commented:
added compatibility for 6.1 and 6.6 kernel

Commit:
Restored Readme

ycurbeloquside commented:
Hi there,

I had tried in Ubuntu 22.04.4 LTS with 6.5.0 kernel version and is not working.

![Screenshot from 2024-07-31 11-33-20](https://github.com/user-attachments/assets/38cd4aef-856a-42cb-8e13-84bf10f0381c)

It is the same error that I am facing when using the master branch.

Am I doing something wrong? Did you test it with this kernel version?
